project/src/main/java/com/google/sps/servlets/CompetitorsServlet.java

```diff
@@ -0,0 +1,112 @@
+        DataSource pool = (DataSource) request.getServletContext().getAttribute("db-connection-pool");
+
+        try (Connection conn = pool.getConnection()) {
+            int competitionId = Integer.parseInt(request.getParameter("competition"));
+            List<CompetitorInfo> competitors = getRankedCompetitors(conn, competitionId);
+            Gson gson = new Gson();
+            response.setContentType("application/json");
+            response.getWriter().println(gson.toJson(competitors));
+        } catch (SQLException ex) {
+            LOGGER.log(Level.WARNING, "Error while attempting to fetch competitors.", ex);
+            response.setStatus(500);
+            response.getWriter().write("Unable to successfully fetch competitors.");
+        }
+    }
+
+    /** 
+     * Return a ranked list of competitors in this competition 
+     */
+    private List<CompetitorInfo> getRankedCompetitors(Connection conn, long competitionId) throws SQLException {
+        List<CompetitorInfo> competitors = new ArrayList<>();
+        String stmt = "SELECT participants.rank, participants.rank_yesterday, participants.net_worth, participants.amt_available, "
+        + "users.name, users.email, users.id FROM participants, users WHERE participants.user=users.id AND participants.competition=" 
+        + competitionId + " ORDER BY participants.rank ASC;";
+
+        try (PreparedStatement competitorStmt = conn.prepareStatement(stmt);) {
+            ResultSet rs = competitorStmt.executeQuery();
+            int rank;
+            int rankYesterday;
+            int netWorth;
+            int amtAvailable;
+            String name;
+            String email;
+            long userId;
+            int numInvestments;
+            while (rs.next()) {
+                rank = rs.getInt(1);
+                rankYesterday = rs.getInt(2);
+                netWorth = rs.getInt(3);
+                amtAvailable = rs.getInt(4);
+                name = rs.getString(5);
+                email = rs.getString(6);
+                userId = rs.getInt(7);
+                numInvestments = getNumberInvestments(conn, userId, competitionId);
+                competitors.add(CompetitorInfo.create(name, email, rank, rankYesterday, netWorth, amtAvailable, numInvestments));
+            }
+            return competitors;
+        }
+    }
+
+    /** 
+     * Returns the number of investments the user currently has
+     */
+    private int getNumberInvestments(Connection conn, long userId, long competitionId) throws SQLException {
+        String stmt = "SELECT * FROM investments WHERE user=" + userId + " AND competition=" + competitionId + ";";
+        int num = 0;
+        try (PreparedStatement competitorStmt = conn.prepareStatement(stmt);) {
+            ResultSet rs = competitorStmt.executeQuery();
+            while (rs.next()) {
+                num++;
+            }
+        return num;
+        }
```

Comment:
can this be an SQL? you are retrieving lots ie (*) from DB.

something like select count(*) FROM investmen.....

you should get back the count.